src/main/java/org/terasology/joshariasSurvival/components/ShearableSheepComponent.java

```diff
@@ -0,0 +1,36 @@
+// Copyright 2021 The Terasology Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+package org.terasology.joshariasSurvival.components;
+
+import org.terasology.engine.entitySystem.Component;
+
+public class ShearableSheepComponent implements Component {
+    int clicks = 0;
+    long lastShearingTimestamp = -1;
+    boolean sheared = false;
+
+    public int getClicks() {
+        return clicks;
+    }
+
+    public void setClicks(int clicks) {
+        this.clicks = clicks;
+    }
+
+    public long getLastShearingTimestamp() {
+        return lastShearingTimestamp;
+    }
+
+    public void setLastShearingTimestamp(long lastShearingTimestamp) {
+        this.lastShearingTimestamp = lastShearingTimestamp;
+    }
+
+    public boolean isSheared() {
+        return sheared;
+    }
+
+    public void setSheared(boolean sheared) {
+        this.sheared = sheared;
+    }
```

Comment:
Not entirely sure we need all this.
Historically, we tried to avoid getters and setters in components in order to not encourage _logic_ being put into components as components should be plain data containers (see https://github.com/MovingBlocks/Terasology/wiki/Entity-System-Architecture#components).

Generally, a component only needs to provide public API if it's supposed to be modified by anything other than it's main managing system. In our case the `ShearingSystem` is the main managing system.
So regarding the public API, we might want to think about the following: Do we want to allow anyone else to modify this component at all? If so, should it be possible to modify the component directly or should the managing system take care of change requests on the component aka provide respective trigger events (i.e. see https://github.com/MovingBlocks/Terasology/wiki/Event-Types#trigger-events)? Also, should every aspect of the component be modifiable or only some parts? 

This might also be a general topic for #architecture. However, I'd say you can already start with the considerations specific to the component at hand.